### PR TITLE
Update vim syntax

### PR DESCRIPTION
**Pull request by smortex**
The vim syntax was introduced 17 years ago for syslong-ng 1.6 but did not get updated since then.  It lacks a lot of syntax highlighting.

Update the syntax to match more of the current syslog-ng language.  Move the syntax in a vim directory that can be used with pathogen / plugged and add an autoload file.

While some content could be auto-generated, this is mostly a manual process, and errors might be present.  This PR intend to move forward and allow users to have a better editing experience with their preferred editor and open PR to adjust the syntax highlighting behavior.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by alltilla** [comment]
@kira-syslogng ok to test

File: contrib/vim/syntax/syslog-ng.vim [hunk 1]
```diff
@@ -0,0 +1,75 @@
+" Vim syntax file
+" Language:	syslog-ng: syslog-ng main configuration file (1.5.5a)
+" Maintainer:	--
+" Last change:	2001 Apr 13
+" URL:		--
+" syslog-ng's home:	http://www.balabit.hu
```
[the remark this answers is not in the document]
Comment (bazsi) [reply contrib/vim/syntax/syslog-ng.vim:6]: yes, please go ahead and change it. The best location atm is I think this github project page.

**Comment by MrAnno** [comment]
@smortex Thanks for this PR.

Please list the newly added files [here](https://github.com/syslog-ng/syslog-ng/blob/master/contrib/Makefile.am); that will make a few CI checks green.

**Comment by smortex** [comment]
`./dbld/rules rpm-centos-7` and `./dbld/rules rpm-fedora-37` seems to pass on my Linux machine \o/

image

**Review by MrAnno — APPROVED** [review]
LGTM.

@czanik Could you check the RPM changes, please? :)

**Comment by czanik** [comment]
I must admit that I do not use vim, but from the RPM point of view it looks OK to me.

**Pull request merged**